src/test/java/io/jenkins/plugins/analysis/warnings/IssueRecorderXmlApiTest.java

```diff
@@ -0,0 +1,199 @@
+package io.jenkins.plugins.analysis.warnings;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.xpath.XPath;
+import javax.xml.xpath.XPathConstants;
+import javax.xml.xpath.XPathExpressionException;
+import javax.xml.xpath.XPathFactory;
+import java.io.File;
+import java.io.IOException;
+import java.util.Collections;
+
+import org.custommonkey.xmlunit.Diff;
+import org.custommonkey.xmlunit.XMLUnit;
+import org.junit.Test;
+import org.w3c.dom.Document;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+
+import com.gargoylesoftware.htmlunit.xml.XmlPage;
+
+import static io.jenkins.plugins.analysis.core.model.Assertions.*;
+import io.jenkins.plugins.analysis.core.steps.IssuesRecorder;
+import io.jenkins.plugins.analysis.core.steps.ToolConfiguration;
+
+import hudson.model.FreeStyleProject;
+
+/**
+ * Integration test for the issue remote api.
+ *
+ * @author Manuel Hampp
+ */
+public class IssueRecorderXmlApiTest extends IssuesRecorderITest {
+
+    /**
+     * Compares the basic xml api (without parameters) against a control result.
+     */
+    @Test
+    public void assertXmlApiMatchesExpected() {
+        try {
+            // setup project
+            FreeStyleProject project = createJobWithWorkspaceFile("checkstyleregextest.xml");
+            enableWarningsWithCheckstyle(project);
+            String buildNumber = String.valueOf(project.getNextBuildNumber());
+            scheduleBuild(project);
+
+            // get xml result from API
+            XmlPage page = j.createWebClient().goToXml(project.getUrl() + buildNumber + "/checkstyleResult/api/xml");
+            Document test = page.getXmlDocument();
+
+            // remove not comparable nodes
+            test = removeRunSpecificXmlNodes(test);
+
+            // get control document
+            Document control = loadControlDocumentFromFile("checkstyleregextest_output.xml");
+
+            // compare documents
+            Diff diff = XMLUnit.compareXML(control, test);
+
+            // assert that document recieved by api is the same as expected
+            assertThat(diff.identical());
+
+        }
+        catch (IOException | SAXException e) {
```

Comment:
try catch entfernen und `j.createWebClient().goToXml(project.getUrl() + buildNumber + "/checkstyleResult/api/xml")` in Methode auslagern die try-catch hat und im Falle einer Exception einen `AssertionError` wirft.